src/msg_handler.c

```diff
@@ -0,0 +1,388 @@
+            return strlen(response);
+        }
+
+        if (strcmp(next_arg, "START") == 0) {
+            next_arg = strtok(NULL, ":");
+            if (next_arg == NULL) {
+                strncat(response, "NACK:Missing button command", max_len);
+                return strlen(response);
+            }
+
+            if (strcmp(next_arg, "GET?") == 0) {
+                strncat(response, int_button_pressed?"1":"0", max_len);
+                strncat(response, ":", max_len - strlen(response));
+                strncat(response, ext_button_pressed?"1":"0", max_len - strlen(response));
+
+                // Clear button state
+                int_button_pressed = false;
+                ext_button_pressed = false;
+                return strlen(response);
+            }
+
+            strncat(response, "NACK:Invalid button command", max_len);
+            return strlen(response);
+        }
+        strncat(response, "NACK:Invalid button name", max_len);
+        return strlen(response);
+    } else if (strcmp(next_arg, "NOTE") == 0) {
+        next_arg = strtok(NULL, ":");
+        if (next_arg == NULL) {
+            strncat(response, "NACK:Missing note frequency", max_len);
+            return strlen(response);
+        }
+
+        int note_freq;
+
+        if (isdigit((int)next_arg[0])) {
+            note_freq = atoi(next_arg);
+            // bounds check
+            if (note_freq < 0) {
+                strncat(response, "NACK:Invalid note frequency", max_len);
+                return strlen(response);
+            }
+        } else {
+            strncat(response, "NACK:Invalid note frequency", max_len);
+            return strlen(response);
+        }
+
+        next_arg = strtok(NULL, ":");
+        if (next_arg == NULL) {
+            strncat(response, "NACK:Missing note duration", max_len);
+            return strlen(response);
+        }
+
+        int note_dur;
+
+        if (isdigit((int)next_arg[0])) {
+            note_dur = atoi(next_arg);
+            // bounds check
+            if (note_dur < 0) {
+                strncat(response, "NACK:Invalid note duration", max_len);
+                return strlen(response);
+            }
+        } else {
+            strncat(response, "NACK:Invalid note duration", max_len);
+            return strlen(response);
+        }
+
+        // Generate note
+        buzzer_note(note_freq, note_dur);
+
+        strncat(response, "ACK", max_len);
+        return strlen(response);
+    } else if (strcmp(next_arg, "*IDN?") == 0) {
+        strncat(response, "Student Robotics:", max_len);
+        strncat(response, BOARD_NAME_SHORT, max_len - strlen(response));
+        strncat(response, ":", max_len - strlen(response));
+        strncat(response, (const char *)SERIALNUM_BOOTLOADER_LOC, max_len - strlen(response));
+        strncat(response, ":", max_len - strlen(response));
+        strncat(response, FW_VER, max_len - strlen(response));
```

Comment:
I think you can employ compile-time string concatenation here.
```suggestion
        strncat(response, "Student Robotics:" BOARD_NAME_SHORT ":", max_len);
        strncat(response, (const char *)SERIALNUM_BOOTLOADER_LOC, max_len - strlen(response));
        strncat(response, ":" FW_VER, max_len - strlen(response));
```